### ISSUE TITLE
Writing to git config should keep casing

### ISSUE THREAD

**Issue by SamVerschueren** [isomorphic-git/isomorphic-git#1570]
While I was testing [`addRemote`](https://isomorphic-git.org/docs/en/addRemote), I noticed that if I passed in a remote name like `FooBar`, it ended up as `[remote "foobar"]` in the git config.

Look at the [`GitConfig` model](https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/GitConfig.js#L207-L211), I noticed that `toLowerCase()` is used to generate the section path. I believe this is incorrect, and it should be written as it comes in.

For instance, if you run `git config FOO.bar bar` or `git config add MyRemote foo`. Looking in the `.git/config` file, the casing is intact.

Retrieving that value is case insensitive though. So running `git config foo.bar` or `git config FOO.BAR` both result in `bar`.

Now, why is this a problem? If you add an upstream by running `git remote add MyUpstream https://github.com/isomorphic-git/isomorphic-git`, casing does make a difference. In this case, I should be able to run `git fetch MyUpstream`, but not `git fetch myupstream`. For this to work correctly, the section in the `.git/config` should be `[remote "MyUpstream"]` and not `[remote "myupstream"]`. Because in the latter, the `listRemotes()` will return `myupstream` which is incorrect.

So long story short. Writing sections to the config file should not be lowercased (because casing matters in certain scenario's). Reading sections should be case insensitive.

I can send a PR to address this.

**Comment by jcubic** [comment]
Yes, you're correct, the branch name should be case sensitive. I've rejected recently #1568 that attempted to make branch check completely case insensitive.

The fix should be applied to remove toLowerCase() and create better unit tests to check this, possibly removing old unit tests that depend on this behavior.

Do you want to make the change?

**Comment by SamVerschueren** [comment]
Yes, I'll have a look at it and send a PR.
